controllers/snapshotenvironmentbinding_controller.go

```diff
@@ -186,23 +185,17 @@ func (f *linkedRemoteSecretsFinalizer) Finalize(ctx context.Context, obj client.
 		return res, unexpectedObjectTypeError
 	}
 
-	// Get the Environment CR
-	environment := appstudiov1alpha1.Environment{}
-	err := f.client.Get(ctx, types.NamespacedName{Name: snapshotEnvBinding.Spec.Environment, Namespace: snapshotEnvBinding.Namespace}, &environment)
-	if err != nil {
-		return res, unableToFetchEnvironmentError
-	}
-
-	target, err := detectTargetFromEnvironment(ctx, environment)
```

Comment:
Is this function still used somewhere else? Can we run into similar problems like in this finalizer in other workflows, too?

Reply:
Yes it is used on target creation too, but hopefully the situation when SEB created w/out Env will never happen yet (but yes we still need to link them somehow to make 101% sure...)